math/mathmore/inc/Math/Polynomial.h

```diff
@@ -62,7 +62,7 @@ namespace Math {
   */
 
 class Polynomial : public ParamFunction<IParamGradFunction>,
-                   public IGradientOneDim
```

Comment:
This was the reason to have IGradientOneDim since now both IParamGradFunction and IgradientFunctionOneDim derive from a common base class, IBaseFunctonOneDim. 
This should be fine since it is a virtual inheritance from IBaseFunctionOneDim